libs/hdf-converters/src/conveyor-mapper.ts

```diff
@@ -0,0 +1,244 @@
+    string,
+    unknown
+  >;
+  const shamappings = {};
+  _.forEach(toplevel, (value) => {
+    if (_.has(value, 'name')) {
+      const temp: string = _.get(value, 'name[0]') || '';
+      _.set(shamappings, temp, _.get(value, 'sha256'));
+    }
+    if (_.has(value, 'children')) {
+      const temp: Record<string, unknown> = _.get(value, 'children') || {};
+      const arr = childrenfinder(temp);
+      _.forEach(arr, (value) => {
+        const temp1: string = value[0] || '';
+        const temp2: string = value[1] || '';
+        _.set(shamappings, temp1, temp2);
+      });
+    }
+  });
+  return shamappings;
+}
+function determineStatus(score: number): ExecJSON.ControlResultStatus {
+  if (score == 0) {
+    return ExecJSON.ControlResultStatus.Passed;
+  }
+  return ExecJSON.ControlResultStatus.Failed;
+}
+
+function arrayifyObject(
+  output: Record<string, unknown>,
+  mapped: Record<string, unknown>
+): unknown {
+  const res = _.get(output, 'api_response.results') as Record<string, unknown>;
+  const newout: Record<string, unknown>[] = [];
+  _.forEach(res, (value) => {
+    const temp = value as Record<string, unknown>;
+    const temp2: string = _.get(value, 'sha256') || '';
+    _.set(temp, 'filename', _.get(mapped, temp2));
+    const descriptions = _.map(
+      _.get(temp, 'result.sections') as Record<string, unknown>[],
+      (val) =>
+        createDescription(
+          val as Record<string, unknown>,
+          _.get(temp, 'result.score') as number,
+          _.get(temp, 'response.milestones.service_started') as string,
+          _.get(temp, 'response.service_name') as string
+        )
+    );
+    if (descriptions.length == 0) {
+      descriptions.push({
+        status: ExecJSON.ControlResultStatus.Passed,
+        code_desc: 'NA',
+        start_time: _.get(temp, 'response.milestones.service_started') as string
+      });
+    }
+    _.set(temp, 'result.sections', descriptions);
+    newout.push(temp);
+  });
+  const groups = _.groupBy(newout, (value) => {
+    return _.get(value, 'response.service_name');
+  });
+  return groups;
+}
+
+function controlMappingConveyor(): MappedTransform<
+  ExecJSON.Control & ILookupPath,
+  ILookupPath
+> {
+  return {
+    id: {path: 'sha256'},
+    title: {path: 'filename'},
+    desc: {path: 'cve_summary'},
+    impact: {
+      path: 'result.score',
+      transformer: (value) => {
+        return value / 1000;
+      }
+    },
+    refs: [],
+    tags: {
+      nist: DEFAULT_NIST_TAG
+    },
+    source_location: {
+      ref: ''
+    },
+    results: [
+      {
+        path: 'result.sections',
+        status: {path: 'status'},
+        code_desc: {path: 'code_desc'},
+        start_time: {path: 'start_time'}
+      }
+    ]
+  };
+}
+
+export class ConveyorMapper extends BaseConverter {
+  data: Record<string, unknown>;
+  type: string;
+  defaultMapping(): MappedTransform<ExecJSON.Execution, ILookupPath> {
```

Comment:
since we aren't passing in any parameters or anything, we could just provide the 'mappings' object which is specified in baseconverter and get rid of the need to do the manual this.setmappings

Reply:
fixed